python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+        name="WCSFitSkyWcsCatalog",
+        storageClass="ExposureCatalog",
+        dimensions=("instrument", "visit", "skymap", "tract"),
+        multiple=True
+    )
+    outputCatalog = pipeBase.connectionTypes.Output(
+        doc=("Source table with stars used in fit, along with residuals in pixel coordinates and tangent "
+             "plane coordinates and chisq values"),
+        name="WCSFit_usedStars",
+        storageClass="DataFrame",
+        dimensions=("instrument", "skymap", "tract"),
+    )
+
+
+class WCSFitConfig(pipeBase.PipelineTaskConfig,
+                   pipelineConnections=WCSFitConnections):
+    """Configuration for WCSFitTask"""
+    sourceSelector = sourceSelectorRegistry.makeField(
+        doc="How to select sources for cross-matching",
+        default="science"
+    )
+    referenceSelector = pexConfig.ConfigurableField(
+        target=ReferenceSourceSelectorTask,
+        doc="How to down-select the loaded astrometry reference catalog.",
+    )
+    matchRadius = pexConfig.Field(
+        doc="Matching tolerance between associated objects (arcseconds)",
+        dtype=float,
+        default=1.0
+    )
+    minMatches = pexConfig.Field(
+        doc="Number of matches required to keep a source object",
+        dtype=int,
+        default=2
+    )
+    allowSelfMatches = pexConfig.Field(
+        doc="Allow multiple object matches from the same viist",
+        dtype=bool,
+        default=False
+    )
+    sourceFluxType = pexConfig.Field(
+        dtype=str,
+        doc="Source flux field to use in source selection and to get fluxes from the catalog.",
+        default='apFlux_12_0'
+    )
+    systematicError = pexConfig.Field(
+        dtype=float,
+        doc="Systematic error padding for the science images (marcsec)",
+        default=0.0
+    )
+    referenceSystematicError = pexConfig.Field(
+        dtype=float,
+        doc="Systematic error padding for the reference catalog (marcsec)",
+        default=0.0
+    )
+    modelComponents = pexConfig.ListField(
+        dtype=str,
+        doc="List of mappings to apply to transform from pixels to sky, in order of their application",
+        default=["INSTRUMENT/DEVICE", "EXPOSURE"]
+    )
+    deviceModel = pexConfig.ListField(
+        dtype=str,
+        doc="List of mappings to apply to transform from detector pixels to intermediate frame",
+        default=["BAND/DEVICE/poly"]
+    )
+    exposureModel = pexConfig.ListField(
+        dtype=str,
+        doc="List of mappings to apply to transform from intermediate frame to sky coordinates",
+        default=["EXPOSURE/poly"]
+    )
+    devicePolyOrder = pexConfig.Field(
+        dtype=int,
+        doc="Order of device polynomial model",
+        default=4
+    )
+    exposurePolyOrder = pexConfig.Field(
+        dtype=int,
+        doc="Order of exposure polynomial model",
+        default=6
+    )
+    useProperMotion = pexConfig.Field(
```

Comment:
I think this should be `doFitProperMotion` not `useProperMotion` which I assume is "used" when `requireProperMotion` is set below?

Reply:
`requireProperMotion` sets a config parameter for the reference object loader, but I had a misunderstanding about what it did, which is just to check that the proper motion parameters are in the schema, not that they are actually there. I've changed this so that 'useProperMotion' just also sets `requireProperMotion`. But I agree that the name should be changed.

Reply:
I changed it to `fitProperMotion`. Let me know if you feel strongly about `doFitProperMotion`.